File: web/src/types/event.ts
export interface EventLocation {
  address: string;
  latitude?: number;
  longitude?: number;
}

export interface Event {
  id: string;
  name: string;
  description?: string;
  date: number;
  imageUrl?: string;
  bannerUrl?: string;
  videoUrl?: string;
  location?: EventLocation;
  city?: string;
  venueName?: string;
  minPrice?: number | null;
  category?: string[];
  tags?: string[];
  eventType?: 'physical' | 'online' | 'hybrid';
  status?: 'draft' | 'submitted' | 'approved' | 'published' | 'rejected' | 'cancelled' | 'active' | 'finished';
  organizerId?: string;
  capacity?: number;
  ticketTypes?: Record<
    string,
    {
      price: number;
      available: number;
      quantity?: number;
    }
  >;
}

export interface RejectedEvent extends Event {
  reason: string;
}

export interface Destination {
  name: string;
  query: string;
  eventCount: number;
}

export interface EventWeather {
  tempC?: number;
  condition?: string;
  icon?: string;
  humidity?: number;
  windKph?: number;
  forecast?: string;
}

File: mobile-organizer/app/src/main/java/com/tdtuer/eventing_organizer/domain/model/Event.kt
package com.tdtuer.eventing_organizer.domain.model

data class Event(
    val bannerUrl: String = "",
    val cancelledAt: Long = 0L,
    val category: List<String> = emptyList(),
    val city: String = "",
    val createdAt: Long = 0L,
    val coordinates: String = "",
    val date: Long = 0L,
    val description: String = "",
    val endDate: Long? = null,
    val eventType: String = "physical",
    val featuredProfiles: List<Any> = emptyList(),
    val geohash: String = "",
    val hotScore: Double = 0.0,
    val id: String = "",
    val imageUrl: String = "",
    val isOutdoor: Boolean = false,
    val lastUpdatedAt: Long = 0L,
    val location: String = "",  // lat: Double, long: Double may crash in the future
    val minPrice: Double? = null,
    val name: String = "",
    val onlineUrl: String = "",
    val recurringRule: Map<String, Any> = emptyMap(),
    val requiredAge: Long = 0L,
    val sponsors: List<String> = emptyList(),
    val status: String = "",
    val tags: List<String> = emptyList(),
    val ticketTypes: Map<String, Map<String, Any>> = emptyMap(),  // vip: {price: Double, quantity: Int, available: Int}
    val venueDetails: Map<String, Any> = emptyMap(),
    val venueName: String = "",
    val videoUrl: String = "",
    val viewCount: Long = 0L,
    val visibility: String = ""
)
